src/algorithms/associations/a_priori.py

```diff
@@ -62,11 +102,23 @@ def _generate_item_sets(self, frequent_sets: Optional[List[tuple]]) -> List[tupl
 
         new_item_sets = []
         for frequent_set_1, frequent_set_2 in combinations(frequent_sets, 2):
-            if not (frequent_set_1[:-1] == frequent_set_1[:-1] and frequent_set_1[-1] < frequent_set_2[-1]):
+            if not (frequent_set_1[:-1] == frequent_set_2[:-1] and frequent_set_1[-1] < frequent_set_2[-1]):
                 continue
 
             new_item_set = self.join(frequent_set_1, frequent_set_2)
-            if not self._has_infrequent_subsets(new_item_set, frequent_sets):
+
+            if with_steps:
+                self.saved_steps.append(
+                    {
+                        "part": APrioriPartLabel.JOIN_AND_PRUNE,
+                        "set 1": frequent_set_1,
+                        "set 2": frequent_set_2,
+                        "new set": new_item_set,
```

Comment:
```suggestion
                        "set_1": frequent_set_1,
                        "set_2": frequent_set_2,
                        "new_set": new_item_set,
```
You use snake_case for other keys, so may change it for consistency.